src/ORM/FieldType/DBUrl.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace SilverStripe\ORM\FieldType;
+
+use SilverStripe\ORM\FieldType\DBVarchar;
+use SilverStripe\Core\Validation\FieldValidation\UrlFieldValidator;
+use SilverStripe\Forms\FormField;
+use SilverStripe\Forms\UrlField;
+
+class DBUrl extends DBVarchar
+{
+    private static array $field_validators = [
+        UrlFieldValidator::class,
```

Comment:
There's a bunch of options in that validator such as what schemes it allows and whether to allow relative protocols - I could see it being very useful to allow those to be defined in the db array.
e.g
```php
private static array $db = [
    'Url' => 'Url(protocols:["http", "https", "ftp"])'
];
```
I probably got the syntax wrong, but you get the idea.

Those could be passed into both the `UrlFieldValidator` here, and also set as appropriate in the `UrlField` in `scaffoldFormField`.

Reply:
Has been split of as separate card https://github.com/silverstripe/silverstripe-framework/issues/11445